app/assets/v2/js/pages/fulfill_bounty.js

```diff
@@ -0,0 +1,120 @@
+
+window.onload = function(){
+    //a little time for web3 injection
+    setTimeout(function(){
+        var account = web3.eth.accounts[0];
+
+        if (typeof localStorage['githubUsername'] !='undefined'){
+            $('input[name=githubUsername]').val(localStorage['githubUsername']);
+        }
+        if (typeof localStorage['notificationEmail'] !='undefined'){
+            $('input[name=notificationEmail]').val(localStorage['notificationEmail']);
+        }
+        if (typeof localStorage['acceptTOS'] !='undefined' && localStorage['acceptTOS']){
+            $('input[name=terms]').attr('checked','checked');
+        }
+        if(getParam('source')){
+            $('input[name=issueURL]').val(getParam('source'));
+        }
+
+
+        $('#submitBounty').click(function(e){
+            mixpanel.track("Claim Bounty Clicked", {});
+            loading_button($('#submitBounty'));
+            e.preventDefault();
+            var notificationEmail = $('input[name=notificationEmail]').val();
+            var githubUsername = $('input[name=githubUsername]').val();
+            var issueURL = $('input[name=issueURL]').val();
+            var claimee_metadata = JSON.stringify({
+                notificationEmail : notificationEmail,
+                githubUsername : githubUsername,
+            });
+            localStorage['githubUsername'] = githubUsername;
+
+            var isError = false;
+            if($('#terms:checked').length == 0){
+                _alert({ message: "Please accept the terms of service." });
+                isError = true;
+            } else {
+                localStorage['acceptTOS'] = true;
+            }
+            if(issueURL == ''){
+                _alert({ message: "Please enter a issue URL." });
+                isError = true;
+            }
+            if(isError){
+                unloading_button($('#submitBounty'));
+                return;
+            }
+
+            var bounty = web3.eth.contract(bounty_abi).at(bounty_address());
+
+            ipfs.ipfsApi = IpfsApi({host: 'ipfs.infura.io', port: '5001', protocol: "https", root:'/api/v0'});
```

Comment:
it might be good to put this information in shared.js so we DRY

Reply:
Hmm... Good point I can try moving the bounty and ipfs infromation out to `shared.js` or somewhere else common.